python/arcticdb/version_store/_store.py

```diff
@@ -1699,11 +1702,16 @@ def _postprocess_df_with_only_rowcount_idx(self, read_result, row_range):
             version=read_result.version.version,
             metadata=meta,
             host=self.env,
-            timestamp=read_result.version.timestamp
+            timestamp=read_result.version.timestamp,
         )
 
     def _resolve_empty_columns(self, columns, implement_read_index):
-        if not implement_read_index and columns == []:
+        is_columns_empty = (
+            columns is None
```

Comment:
This is used in order to implement [this](https://github.com/man-group/ArcticDB/blob/f9d955f71fbb7439ddce8724dda5ec8e23f6c171/python/arcticdb/version_store/library.py#L1445) in the NativeVersionStore [here](https://github.com/man-group/ArcticDB/blob/f9d955f71fbb7439ddce8724dda5ec8e23f6c171/python/arcticdb/version_store/_store.py#L1754)

None should mean read all columns while empty list should mean read only the index. I don't think this will work correctly but the tests are passing. Either I'm missing something or we have bad tests.

Reply:
No idea about that, my change is mainly for supporting the case where `columns` is `ndarray` (`isinstance(columns, np.ndarray) and columns.size == 0)`) because some of the tests pass a `numpy array` to `columns` and `Numpy 2` is not happy with the old check.